cloud/misc/ovirt.py

```diff
@@ -214,110 +214,168 @@
-        # define disk params
-        vmdisk= params.Disk(size=1024 * 1024 * 1024 * int(vmdisk_size), wipe_after_delete=True, sparse=False, interface=vmdisk_int, type_="System", format='raw',
-        storage_domains=params.StorageDomains(storage_domain=[conn.storagedomains.get(name=sdomain)]))
-        # define network parameters
-        network_net = params.Network(name=vmnetwork)
-        nic_net1 = params.NIC(name=vmnic, network=network_net, interface='virtio')
-        
-    try:
-        conn.vms.add(vmparams)
-    except:
-        print "Error creating VM with specified parameters"
-        sys.exit(1)
-    vm = conn.vms.get(name=vmname)
-    try:
-        vm.disks.add(vmdisk)
-    except:
-        print "Error attaching disk"
-    try:
-        vm.nics.add(nic_net1)
-    except:
-        print "Error adding nic"
-
-
-# create an instance from a template
-def create_vm_template(conn, vmname, image, zone):
-    vmparams = params.VM(name=vmname, cluster=conn.clusters.get(name=zone), template=conn.templates.get(name=image),disks=params.Disks(clone=True))
-    try:
-        conn.vms.add(vmparams)
-    except:
-        print 'error adding template %s' % image
-        sys.exit(1)
-
-
-# start instance
-def vm_start(conn, vmname):
-    vm = conn.vms.get(name=vmname)
-    vm.start()
-
-# Stop instance
-def vm_stop(conn, vmname):
-    vm = conn.vms.get(name=vmname)
-    vm.stop()
-
-# restart instance
-def vm_restart(conn, vmname):
-    state = vm_status(conn, vmname)
-    vm = conn.vms.get(name=vmname)
-    vm.stop()
-    while conn.vms.get(vmname).get_status().get_state() != 'down':
-        time.sleep(5)
-    vm.start()
-
-# remove an instance
-def vm_remove(conn, vmname):
-    vm = conn.vms.get(name=vmname)
-    vm.delete()
-
-# ------------------------------------------------------------------- #
-# VM statuses
-#
-# Get the VMs status
-def vm_status(conn, vmname):
-    status = conn.vms.get(name=vmname).status.state
-    print "vm status is : %s" % status
-    return status
-
-
-# Get VM object and return it's name if object exists
-def get_vm(conn, vmname):
-    vm = conn.vms.get(name=vmname)
-    if vm == None:
-        name = "empty"
-        print "vmname: %s" % name
-    else:
-        name = vm.get_name()
-        print "vmname: %s" % name
-    return name
+        storage_domains = params.StorageDomains(storage_domain=[self.conn.storagedomains.get(name=sdomain)])
+        if vmdisk_alloc == 'thin':
+            # define disk params
+            vmdisk= params.Disk(
+                size=1024 * 1024 * 1024 * int(vmdisk_size),
+                wipe_after_delete=True,
+                sparse=True,
+                interface=vmdisk_int,
+                type_="System",
+                format='cow', 
+                bootable=True,
+                storage_domains=storage_domains)
+        elif vmdisk_alloc == 'preallocated':
```

Comment:
if the olny difference here is the sparse parameter, it seems cleaner to just populate a var with it and assign